File: LeetCode/791自定义字符串排序.go
func customSortString(order string, s string) (ans string) {
	str := []byte(s)
	for i := 0; i < len(order); i++ {
		for j := 0; j < len(str); j++ {
			if str[j] == order[i] {
				ans += string(str[j])
				str[j] = '.'
			}
		}
	}
	for i := 0; i < len(str); i++ {
		if str[i] == '.' {
			continue
		}
		ans += string(str[i])
	}
	return
}

File: LeetCode/791自定义字符串排序.java
class Solution {
    public String customSortString(String order, String s) {
        StringBuffer ans = new StringBuffer();
        char[] str = s.toCharArray();
        for (int i = 0; i < order.length(); i++) {
            for (int j = 0; j < str.length; j++) {
                if (order.charAt(i) == str[j]) {
                    ans.append(str[j]);
                    str[j] = '.';
                }
            }
        }
        for (int i = 0; i < str.length; i++) {
            if (str[i] == '.') continue;
            ans.append(str[i]);
        }
        return ans.toString();
    }
}